internal/metrics/metrics.go

```diff
@@ -0,0 +1,68 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// When adding metric names, see https://prometheus.io/docs/practices/naming/#metric-names
+const (
+	registeredEdgeDevicesQuery                   = "k4e_operator_edge_devices_were_registered"
+	failedToCompleteRegistrationEdgeDevicesQuery = "k4e_operator_edge_devices_failed_to_register"
+	createdOBCQuery                              = "k4e_operator_created_obc"
+)
+
+var (
+	registeredEdgeDevices = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: registeredEdgeDevicesQuery,
+			Help: "Number of registered EdgeDevices",
+		},
+	)
+	failedToCompleteRegistrationEdgeDevices = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: failedToCompleteRegistrationEdgeDevicesQuery,
+			Help: "Number of failed registration EdgeDevices",
+		},
+	)
+	createdOBC = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: createdOBCQuery,
+			Help: "Number of created OBCs",
+		},
+	)
+)
+
+func init() {
+	// Register custom metrics with the global prometheus registry
+	metrics.Registry.MustRegister(
+		registeredEdgeDevices,
+		failedToCompleteRegistrationEdgeDevices,
+		createdOBC,
+	)
+}
+
+//go:generate mockgen -source=metrics.go -package=metrics -destination=mock_metrics_api.go
+
+// Metrics is an interface representing a prometheus client for the Special Resource Operator
+type Metrics interface {
+	IncRegisteredEdgeDevice()
+	IncFailedRegistrationEdgeDevice()
+	IncCreatedOBCs()
+}
+
+func New() Metrics {
+	return &metricsImpl{}
+}
+
+type metricsImpl struct{}
+
+func (m *metricsImpl) IncRegisteredEdgeDevice() {
```

Comment:
Is this metric supposed to represent just number of registrations or current number of edgedevices under operator's purview?

Reply:
Just number of registrations

Reply:
In that case the name of the method name is a little confusing to me - it can mean both. WDYT about making the name a bit more explicit; like `IncEdgeDeviceRegisteredSuccessfully` and `k4e_operator_edge_devices_registered_successfully` or `IncEdgeDeviceSuccessfulRegistration` and `k4e_operator_edge_devices_successful_registration` or simply aligning it with the metric name: `IncEdgeDevicesWereRegistered`.